e2e/main_test.go

```diff
@@ -0,0 +1,134 @@
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// global client for use by all tests
+var client kubernetes.Interface
+
+// non-configurable for now
+const namespace = "bootkubetesting"
+
+// TestMain handles setup before all tests
+func TestMain(m *testing.M) {
+	var (
+		kubeconfig    = flag.String("kubeconfig", "../hack/quickstart/cluster/auth/kubeconfig", "absolute path to the kubeconfig file")
+		expectedNodes = flag.Int("nodes", 0, "the number of nodes to expect")
```

Comment:
I figured using default values here is always incorrect. Expected nodes should be explicit. So there is no safe default. Instead, set it to an obviously incorrect sentinel that will cause an error message before tests start.

Also setting it to 1 and not having 1 node would cause flakeyness (if two nodes check in before 1 is ready).  We could have 0 disable the check altogether, but at that point if we support that we should probably remove the check altogether adjust timeouts among tests appropriately.